INSTALL-cross-linux-arm64.md

```diff
@@ -0,0 +1,228 @@
+#set(JSONC_LIBRARY           "/usr/lib64/libjson-c.a")
+
+#
+# Uncomment these as needed:
+#
+#set(JSONC_INCLUDE_DIR       "/usr/include/json-c")
+#set(JSONC_LIBRARY           "/usr/lib64/libjson-c.a")
+#set(ENABLE_JSON_SHARED      OFF)
+
+#set(BZIP2_INCLUDE_DIR       "/usr/include/")
+#set(BZIP2_LIBRARY           "/usr/lib64/libbz2.a")
+
+#set(OPENSSL_ROOT_DIR        "/usr/")
+#set(OPENSSL_INCLUDE_DIR     "/usr/include/")
+#set(OPENSSL_CRYPTO_LIBRARY  "/usr/lib64/libcrypto.so")
+#set(OPENSSL_SSL_LIBRARY     "/usr/lib64/libssl.so")
+
+#set(LIBXML2_INCLUDE_DIR     "/usr/include/libxml2")
+#set(LIBXML2_LIBRARY         "/usr/lib64/libxml2.so")
+
+#set(PCRE2_INCLUDE_DIR       "/usr/include/")
+#set(PCRE2_LIBRARY           "/usr/lib64/libpcre2-8.so")
+
+#set(CURSES_INCLUDE_DIR      "/usr/include/")
+#set(CURSES_LIBRARY          "/usr/lib64/libncurses.a")
+
+#set(ZLIB_INCLUDE_DIR        "/usr/include/")
+#set(ZLIB_LIBRARY            "/usr/lib64/libz.so")
+
+#set(LIBCHECK_INCLUDE_DIR    "/usr/include/")
+#set(LIBCHECK_LIBRARY        "/usr/lib64/libcheck.a")
+```
+
+### If not using a sysroot
+
+Without a sysroot, you must tell CMake exactly where to find the library dependencies built for aarch64.
+
+> _IMPORTANT_: Without a sysroot, your runtime platform must have these EXACT SAME libraries.
+
+`CMAKE_TOOLCHAIN_ARM64.cmake`:
+```cmake
+# Platform
+set(CMAKE_SYSTEM_NAME       Linux)
+set(CMAKE_SYSTEM_PROCESSOR  arm64)
+set(CMAKE_C_COMPILER        "aarch64-linux-gnu-gcc")
+set(CMAKE_CXX_COMPILER      "aarch64-linux-gnu-g++")
+set(RUST_COMPILER_TARGET    "aarch64-unknown-linux-gnu")
+
+# Project Variables needed to cross compile
+set(HAVE_ATTRIB_ALIGNED     1)
+set(HAVE_ATTRIB_PACKED      1)
+set(HAVE_UNAME_SYSCALL      1)
+set(HAVE_SAR                1)
+set(HAVE_FD_PASSING         1)
+set(MMAP_FOR_CROSSCOMPILING ON)
+set(ENABLE_SYSTEMD          OFF)
+
+set( test_run_result
+     "PLEASE_FILL_OUT-FAILED_TO_RUN"
+     CACHE STRING "Result from try_run" FORCE)
+
+set( test_run_result__TRYRUN_OUTPUT
+     "PLEASE_FILL_OUT-NOTFOUND"
+     CACHE STRING "Output from try_run" FORCE)
+
+#
+# Dependencies
+#
+
+set(JSONC_INCLUDE_DIR       "/usr/include/json-c")
+set(JSONC_LIBRARY           "/usr/lib/aarch64-linux-gnu/libjson-c.a")
+set(ENABLE_JSON_SHARED      OFF)
+
+set(BZIP2_INCLUDE_DIR       "/usr/include/")
+set(BZIP2_LIBRARY           "/usr/lib/aarch64-linux-gnu/libbz2.a")
+
+set(OPENSSL_ROOT_DIR        "/usr/")
+set(OPENSSL_INCLUDE_DIR     "/usr/include/")
+set(OPENSSL_CRYPTO_LIBRARY  "/usr/lib/aarch64-linux-gnu/libcrypto.so")
+set(OPENSSL_SSL_LIBRARY     "/usr/lib/aarch64-linux-gnu/libssl.so")
+
+set(LIBXML2_INCLUDE_DIR     "/usr/include/libxml2")
+set(LIBXML2_LIBRARY         "/usr/lib/aarch64-linux-gnu/libxml2.so")
+
+set(PCRE2_INCLUDE_DIR       "/usr/include/")
+set(PCRE2_LIBRARY           "/usr/lib/aarch64-linux-gnu/libpcre2-8.so")
+
+set(CURSES_INCLUDE_DIR      "/usr/include/")
+set(CURSES_LIBRARY          "/usr/lib/aarch64-linux-gnu/libncurses.a")
```

Comment:
I received link errors for stuff in 'ncurses'.  Changing to the shared object resolved them.  I believe we also need 'tinfo' for static linking.

Reply:
Ha looks like you found the same thing about libtinfo that I commented on a minute prior.

Reply:
Good to know that using the `.so` also resolves it.